samples/MvcApp/App_Start/Startup.Auth.cs

```diff
@@ -23,37 +21,30 @@ public void ConfigureAuth(IAppBuilder app)
 
             // These must match the data protection settings in MvcCoreApp Program.cs for cookie sharing to work
             var sharedApplicationName = "CommonMvcAppName";
-            var sharedDataProtectionProvider = DataProtectionProvider.Create(
-                // This directory is used to share dataprotection keys between MvcApp and MvcCoreApp
-                new DirectoryInfo(Path.Combine(Path.GetTempPath(), "sharedkeys", sharedApplicationName)),
-                    builder => builder.SetApplicationName(sharedApplicationName))
-                    .CreateProtector(
-                        "Microsoft.AspNetCore.Authentication.Cookies.CookieAuthenticationMiddleware",
-                        // Must match the Scheme name on the MvcCoreApp, i.e. IdentityConstants.ApplicationScheme
-                        "SharedCookie",
-                        "v2");
+            var sharedCookieName = ".AspNet.ApplicationCookie";
+            // This directory is used to share dataprotection keys between MvcApp and MvcCoreApp
+            var sharedKeyDirectory = Path.Combine(Path.GetTempPath(), "sharedkeys", sharedApplicationName);
+            // Must match the Scheme name on the MvcCoreApp, i.e. IdentityConstants.ApplicationScheme
+            var authScheme = "SharedCookie";
 
             // Enable the application to use a cookie to store information for the signed in user
             // and to use a cookie to temporarily store information about a user logging in with a third party login provider
             // Configure the sign in cookie
-            app.UseCookieAuthentication(new CookieAuthenticationOptions
-            {
-                AuthenticationType = DefaultAuthenticationTypes.ApplicationCookie,
-                LoginPath = new PathString("/Account/Login"),
-                Provider = new CookieAuthenticationProvider
+            app.UseCookieAuthentication(app.ConfigureSharedCookie(sharedApplicationName, sharedCookieName, authScheme, sharedKeyDirectory,
+                new CookieAuthenticationOptions
                 {
-                    // Enables the application to validate the security stamp when the user logs in.
-                    // This is a security feature which is used when you change a password or add an external login to your account.  
-                    OnValidateIdentity = SecurityStampValidator.OnValidateIdentity<ApplicationUserManager, ApplicationUser>(
-                        validateInterval: TimeSpan.FromMinutes(30),
-                        regenerateIdentity: (manager, user) => user.GenerateUserIdentityAsync(manager))
-                },
-
-                // Settings to configure shared cookie with MvcCoreApp
-                CookieName = ".AspNet.ApplicationCookie",
```

Comment:
Might as well keep setting the cookie name directly, putting it in the configure method doesn't make it easier.